### PR TITLE
Add documentation for asgardeo auth provider

### PR DESCRIPTION
**Pull request by yathindrakodithuwakku**
## Changes 💡
Add Asgardeo setup documentation.

## Affected issues 🎟
https://github.com/nextauthjs/next-auth/pull/3674

### CODE REVIEW

File: docs/providers/asgardeo.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+https://wso2.com/asgardeo/docs/guides/authentication
+
+
+## Example
+
+### To allow specific Active Directory users access:
+
+- Log into https://console.asgardeo.io.
+- Next, go to "Application" in https://console.asgardeo.io/t/{your_tenant}/develop/applications .
```
Comment (senthalan) [inline docs/providers/asgardeo.md:16]: Isn't it better if we can guide this step through UI? Other than providing this URL. We can explain the flow in summary and link to the Asgardeo application creation doc for more information.

File: docs/providers/asgardeo.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+- Register standard based - Open id connect, application
+- After registring the application, go to protocol tab.
+- Check client credentials grant type.
```
Comment (senthalan) [inline docs/providers/asgardeo.md:19]: Do we need `client credentials` grant type ? isn't `authorization code `

File: docs/providers/asgardeo.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+- Add Authorized redirect URLs & Allowed origins fields.
+
+In `.env` add the following entries:
+
+```
+NEXTAUTH_URL=<Your application url e.g: http://localhost:3000>
+SECRET=<Your secret here>
+ASGARDEO_ORGANIZATION=<copy the organization name(tenant) here>
```
Comment (senthalan) [inline docs/providers/asgardeo.md:27]: Better not to specify tenant. Only having `organization name` should be fine.
```suggestion
ASGARDEO_ORGANIZATION=<copy the organization name here>
```

File: docs/providers/asgardeo.md (hunk 2 above)
Comment (senthalan) [reply docs/providers/asgardeo.md:19]: can't we guide to use the single page application template?

File: docs/providers/asgardeo.md (hunk 1 above)
Comment (yathindrakodithuwakku) [reply docs/providers/asgardeo.md:16]: yes that would be great. Will link the guide.

File: docs/providers/asgardeo.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+---
+
+## Documentation
+
+https://wso2.com/asgardeo/docs/guides/authentication
+
+
+## Example
+
+### To allow specific Active Directory users access:
```
Comment (mefarazath) [inline docs/providers/asgardeo.md:13]: Are these intended?

Comment (yathindrakodithuwakku) [reply docs/providers/asgardeo.md]: Removed the invalid title

**Comment by ThangHuuVu** [comment]
Thanks! NextAuth.js has moved on to the monorepo structure, so could you please include this change to your PR there? https://github.com/nextauthjs/next-auth/pull/3674

**Comment by yathindrakodithuwakku** [comment]
@ThangHuuVu Closing this pr as the doc content has been moved to https://github.com/nextauthjs/next-auth/pull/3674

**Pull request closed without merging**